MSAL/src/public/MSALError.h

```diff
@@ -117,235 +115,223 @@ extern NSString *MSALBrokerVersionKey;
-    MSALErrorAmbiguousAuthority     = -42201,
-    MSALErrorUserNotFound           = -42202,
-    MSALErrorNoAccessTokensFound    = -42203,
-    MSALErrorWrapperCacheFailure    = -42270,
-    /*!
-        MSAL encounted a network error while trying to authenticate. Inspect
-        NSUnderlyingError from the userInfo dictionary for more information
-        about the specific error. In most cases the errors will come from the
-        system's network layer and the individual errors will be detailed in
-        Apple's <Foundation/NSURLError.h> header file.
+     A required parameter was not provided, or a passed in parameter was
+     invalid. See MSALErrorDescriptionKey for more information.
      */
-    MSALErrorNetworkFailure = -42300,
+    MSALErrorInvalidParameter                   = -42000,
     
     /*!
-        The user cancelled the web auth session by tapping the "Done" button on the
-        SFSafariViewController.
-     */
-    MSALErrorUserCanceled = -42400,
-    /*!
-        The authentication request was cancelled programmatically.
-     */
-    MSALErrorSessionCanceled = -42401,
-    /*!
-        An interactive authentication session is already running with the
-        SafariViewController visible. Another authentication session can not be
-        launched yet.
+     The required MSAL URL scheme is not registered in the app's info.plist.
+     The scheme should be "msal<clientid>"
+     
+     e.g. an app with the client ID "abcde-12345-vwxyz-67890" would need to
+     register the scheme "msalabcde-12345-vwxyz-67890" and add the
+     following to the info.plist file:
+     
+     <key>CFBundleURLTypes</key>
+     <array>
+     <dict>
+     <key>CFBundleTypeRole</key>
+     <string>Editor</string>
+     <key>CFBundleURLName</key>
+     <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
+     <key>CFBundleURLSchemes</key>
+     <array>
+     <string>msalabcde-12345-vwxyz-67890</string>
+     </array>
+     </dict>
+     
      */
-    MSALErrorInteractiveSessionAlreadyRunning = -42402,
+    MSALErrorRedirectSchemeNotRegistered        = -42001,
+    
     /*!
-        MSAL could not find the current view controller in the view controller
-        heirarchy to display the SFSafariViewController on top of.
+     The passed in authority URL does not pass validation.
+     If you're trying to use B2C, you must disable authority validation by
+     setting validateAuthority of MSALPublicClientApplication to NO.
      */
-    MSALErrorNoViewController = -42403,
+    MSALErrorFailedAuthorityValidation          = -42010,
+    
+    MSALErrorMismatchedUser                     = -42101,
     
     /*!
-        MSAL tried to open a URL from an extension, which is not allowed.
+     MSAL requires a non-nil account for the acquire token silent call
      */
-    MSALErrorAttemptToOpenURLFromExtension = -42404,
-
+    MSALErrorAccountRequired                    = -42106,
+    
+    MSALErrorCacheBadFormat                     = -42270,
```

Comment:
will this one ever happen?

Reply:
Right, it is for mac cache only, removed it.